tests/models/test_ngsi_v2_iot.py

```diff
@@ -214,6 +214,142 @@ def test_expression_language(self):
 
             assert len(w) == 2
 
+    def test_add_device_attributes(self):
+        """
+        Test the device model regarding the behavior with devices attributes.
+        According to https://iotagent-node-lib.readthedocs.io/en/latest/api.html and
+        based on our best practice, following rules are checked
+            - name is required, but not necessarily unique
+            - object_id is not required, if given must be unique, i.e. not equal to any
+                existing object_id and name
+        """
+        def initial_device():
+            attr = DeviceAttribute(
+                name="temperature",
+                type="Number"
+            )
+            return Device(
+                device_id="dummy:01",
+                entity_name="entity:01",
+                entity_type="MyEntity",
+                attributes=[attr]
+            )
+
+        # fail, because attr1 and attr ara identical
+        device_a = initial_device()
+        attr1 = DeviceAttribute(
+            name="temperature",
+            type="Number"
+        )
+        with self.assertRaises(ValueError):
+            device_a.add_attribute(attribute=attr1)
+
+        # fail, because the object_id is duplicated with the name of attr1
+        device_b = initial_device()
+        attr2 = DeviceAttribute(
```

Comment:
The name must not be unique, right? Why is object_id compared with name of attr1?